docs/content/docs/connectors/datastream/formats/csv.md

```diff
@@ -0,0 +1,87 @@
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+# CSV format
+
+To use the CSV format you need to add the Flink CSV dependency to your project:
+
+```xml
+<dependency>
+	<groupId>org.apache.flink</groupId>
+	<artifactId>flink-csv</artifactId>
+	<version>{{< version >}}</version>
+</dependency>
+```
+
+### Read
+
+Flink supports reading CSV files using `CsvReaderFormat`. The reader utilizes Jackson library and allows passing the corresponding configuration for the CSV schema and parsing options.
+
+`CsvReaderFormat` can be initialized and used like this:
+```java
+CsvReaderFormat<SomePojo> csvFormat = CsvReaderFormat.forPojo(SomePojo.class);
+FileSource<SomePojo> source = 
+        FileSource.forRecordStreamFormat(csvFormat, Path.fromLocalFile(...)).build();
+```
+
+The schema for CSV parsing, in this case, is automatically derived based on the fields of the `SomePojo` class using the `Jackson` library. (Note: you might need to add `@JsonPropertyOrder({field1, field2, ...})` annotation to your class definition with the fields order exactly matching those of the CSV file columns).
+
+If you need more fine-grained control over the CSV schema or the parsing options, use the more low-level `forSchema` static factory method of `CsvReaderFormat`:
+
+```java
+CsvReaderFormat<T> forSchema(CsvMapper mapper, 
+                             CsvSchema schema, 
+                             TypeInformation<T> typeInformation) 
+```
+
+Similarly to the `TextLineInputFormat`, `CsvReaderFormat` can be used in both continues and batch modes (see [TextLineInputFormat]({{< ref "docs/connectors/formats/text_files" >}}  for examples).
+
+### Write
+
+For writing CSV files, use `CsvBulkWriter`. An example of producing CSV files from a stream of POJOs  looks like this:
+
+```java
+DataStream<SomePojo> stream = ...
+BulkWriter.Factory<SomePojo> bulkWriterFactory = (out) -> CsvBulkWriter.forPojo(SomePojo.class, out);
+
+FileSink<SomePojo> sink = FileSink.forBulkFormat(new Path(...), bulkWriterFactory)
+                                    .withBucketAssigner(new BasePathBucketAssigner<>())
+                                    .build();
+
+stream.sinkTo(sink);
+```
+
+Similarly to the reader, the writer also utilizes Jackson library and supports fully customizable configuration using its `CsvMapper` and `CsvSchema` classes:
+
+```java
+
+Converter<CityPojo, CityPojo, Void> converter = (value, context) -> value; //An optional custom types converter, no-op in this example 
+CsvMapper csvMapper = new CsvMapper();
+CsvSchema schema = csvMapper.schemaFor(CityPojo.class).withColumnSeparator("|");;
+BulkWriter.Factory<CityPojo> bulkWriterFactory = (out) -> CsvBulkWriter.forSchema(csvMapper, schema, converter, null, out);
```

Comment:
Does this really work? The class `CsvBulkWriter` seems to be package-private. I suppose we are not supporting writing Csv files outside of the Table API?

Reply:
Hmm, indeed. I think Arvid proposed to keep it so, but that was before we added convenience static factory methods. It was not really defined as a requirement, but I do not see why not expose it in the next version.

Reply:
Hmm, indeed. I think Arvid proposed to keep it so, but that was before we added convenience static factory methods. It was not really defined as a requirement, but I do not see why not expose it in the next version.

Reply:
Hmm, indeed. I think Arvid proposed to keep it so, but that was before we added convenience static factory methods. It was not really defined as a requirement, but I do not see why not expose it in the next version.